tests/devices/qubit/test_apply_operation.py

```diff
@@ -389,7 +389,8 @@ def test_large_state_small_matrix_evolves_matrix(self, mocker):
 
         # seems like _evolve_state_vector_under_parametrized_evolution calls
         # einsum twice, and the default apply_operation only once
-        assert spy.call_count == 1
+        # and it seems that getting the matrix from the hamiltonian calls einsum a few times.
+        assert spy.call_count == 6
```

Comment:
If we don't understand so well where these numbers come from, why are we testing them at all?

Reply:
don't know why we were testing it, but now we call einsum more times due to it's use when calculating the matrix.